### ISSUE TITLE
Shared imports between Runestone Components and the Runestone server

### ISSUE THREAD

**Issue by bjones1** [RunestoneInteractive/rs#110]
This continues the discussion in RunestoneInteractive/RunestoneComponents#876 as a separate issue.

Goal: A way to provide common imports shared between Runestone Components and the Runestone server.

Constraints: Because these imports may be used in controllers or models of web2py, they must be minimal. That is, they must avoid imports of unrelated code (such as Runestone Components-only Sphinx `setup()` functions) or expensive operations (creating a database connection) which aren't required for the correct operation of the import.

Uses:
- Replace the `from runestone import cmap` and `from runestone.server.lp_common_lib import STUDENT_SOURCE_PATH, ...` statements in the Runestone server.
- Possibly: share relevant model files, so we can use DAL instead of re-creating models in SQLAlchemy to modify the database in Runestone Components.

Challenges:
- Currently, the Runestone server imports from Runestone Components. This isn't the best strategy, since it ends up importing a lot of components-only symbols.
- The Runestone Components `conf.py` file contains `from runestone import runestone_static_dirs, runestone_extensions`. Since this is now a part of each book, changing it would be painful. The `runestone.__init__` module begins with wildcard imports of all directives, making it a poor candidate for sharing with the server. In addition, it imports some command-line stuff (paver), which also isn't needed by the server.

Approaches:
- Create another package, perhaps called `runestone_common`, so we don't have to rethink/refactor `runestone`. Objection: two packages in one distribution seems a bit messy.
- One alternative: put all such imports in the web2py `modules` directory, which is evaluated only once. Then, we don't really care about performance. But a bit ugly from a modularity perspective.

**Comment by bnmnetp** [comment]
Closing due to old age. likely already fixed or not applicable anymore
